testing/upload_validation.ts

```diff
@@ -2160,6 +2162,12 @@ ${endpointKey ? 'endpointKey is set' : `requiresEndpointUrl is ${requiresEndpoin
     tools: z.array(toolSchema),
   });
 
+  const mcpProviderSchema = zodCompleteObject<McpProvider>({
+    endpoint: z.string().min(1).url('MCP provider endpoint must be a valid URL'),
```

Comment:
Won't this have to be tied into the connection object's endpoint URL so we can handle dynamic domain situations?